lib/src/simple_op_store.rs

```diff
@@ -495,10 +518,16 @@ fn view_from_proto(proto: crate::protos::op_store::View) -> View {
     }
 
     #[allow(deprecated)]
-    if proto.git_head.is_some() {
-        view.git_head = ref_target_from_proto(proto.git_head);
+    if !proto.git_heads.is_empty() {
+        view.git_heads = git_heads_from_proto(proto.git_heads);
+    } else if proto.git_head.is_some() {
+        view.git_heads = hashmap! {
+            WorkspaceId::default() => ref_target_from_proto(proto.git_head),
```

Comment:
Since the "default" workspace could have been renamed, we might have to guess the workspace id based on `wc_commit_ids`.

Reply:
Can you show me how to rename the default workspace? There is no `jj workspace rename` and we don't care about the case where people run `jj workspace forget default` and then use another one, because until this PR, no other workspace than default could have had a git_head created. Does this have to be through using JJ as a library? Is there a config flag I'm not seeing?

I wanted to create a test, and I'm struggling.

Reply:
`jj workspace rename` was added recently. That said, it's good point that non-"default" workspace wouldn't have a colocated git repo.

I was thinking of using the workspace name if `wc_commit_ids.len() == 1` or something, but it might be better to just check if `wc_commit_ids.contains_key("default")`. It could go wrong, but new `git_heads[id]` will be set at some point later.

Reply:
Would it be okay to do a really hacky thing like this in the `view_from_proto` function, on the basis that the default workspace could not actually be in any other location before this PR?

```rust
            let guess_default_workspace_id = store_path
                .and_then(|sp| sp.parent())
                .and_then(|repo| repo.parent())
                .map(|jj| jj.join("working_copy").join("checkout"))
                .and_then(|checkout| fs::read(checkout).ok())
                .and_then(|buf| crate::protos::working_copy::Checkout::decode(buf.as_slice()).ok())
                .map(|checkout| checkout.workspace_id)
                .map(WorkspaceId::new)
                .filter(|workspace_id| view.wc_commit_ids.contains_key(workspace_id));
```

Reply:
In any case it's going to be great to have a CLI test that manipulates the views to make them use deprecated fields and sees what JJ can do with them.

Reply:
> Would it be okay to do a really hacky thing like this in the `view_from_proto` function,

I would avoid that so long as the `git_heads` entry can be recovered by e.g. checking out new commit. And as you said, it's difficult to test.

Reply:
I did manage to write a test for this. Possibly the most long-winded test in the whole repo. But I think we needed something to exercise the deprecated fields.